pkg/apis/numaflow/v1alpha1/mono_vertex_types.go

```diff
@@ -472,10 +472,9 @@ func (mvspec MonoVertexSpec) buildContainers(req getContainerReq) []corev1.Conta
 	if mvspec.Sink.UDSink != nil { // Only support UDSink for now.
 		containers = append(containers, mvspec.Sink.getUDSinkContainer(req))
 	}
-	if mvspec.Sink.Fallback != nil {
+	if mvspec.Sink.Fallback != nil && mvspec.Sink.Fallback.UDSink != nil {
 		containers = append(containers, mvspec.Sink.getFallbackUDSinkContainer(req))
 	}
```

Comment:
don't we support builtins now?

Reply:
we do support, this check is to avoid creating sidecar container when the fallback sink is not udsink

Reply:
gotcha, this was a bug